pkg/cmd/cli/describe/printer.go

```diff
@@ -138,6 +140,11 @@ func NewHumanReadablePrinter(noHeaders, withNamespace, wide bool, showAll bool,
 	return p
 }
 
+func NewHumanReadablePrinterWithResourceAliases(resourceAliases map[string]string, noHeaders, withNamespace, wide bool, showAll bool, showLabels bool, absoluteTimestamps bool, columnLabels []string) *kctl.HumanReadablePrinter {
+	resourceShortForms = resourceAliases
```

Comment:
@rhcarvalho Taking an idea from your suggestions [here](https://github.com/openshift/origin/pull/9334#discussion_r69468339), maybe a clean way to address this would be to edit `pkg/cmd/util/clientcmd/factory.go`, so that instead of [line 379](https://github.com/openshift/origin/pull/9334/files#diff-0073015bc0e88b3269344b5a87d471dbR379) returning the `HumanReadablePrinter` right away, we store it in a `printer` var, and do something similar to:

```go
	// ...
	if mustPrintWithKinds(...) { // mustPrintWithKinds replaces yet another for-loop bloating the over-bloated RunGet.
		printer.EnsurePrintWithKind()  // this method doesn't exist yet, but would set PrintOptions.WithKind to true.
	}
```